File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<int> rowgen(int n){
        long long res=1;
        vector<int>ans;
        ans.push_back(1);
        for(int col=1;col<n;col++){
            res=res*(n-col);
            res=res/col;
            ans.push_back(res);
        }
        return ans;
    }
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>v;
        for(int i=1;i<=numRows;i++){
            vector<int>temp=rowgen(i);
            v.push_back(temp);
        }
        return v;
    }
};

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def rowgen(self,i):
        a=[]
        for j in range(i):
            a.append(comb(i-1,j))
        return a
    def generate(self, numRows: int) -> List[List[int]]:
        res=[]
        for i in range(1,numRows+1):
            res.append(self.rowgen(i))
        return res
